### PR TITLE
3 get/api/topics

### PR DESCRIPTION
**Pull request by valy4**

### CODE REVIEW

File: Controllers/controllers.errors.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+exports.handle404Paths = (req, res, next) => {
+  res.status(404).send({ msg: "Page not found!" });
+};
+
+exports.handle500Errors = (err, req, res, next) => {
+  res.status(500).send({ msg: "Server Error" });
```
Comment (roisinoreilly) [inline Controllers/controllers.errors.js:6]: Nicely handled

File: Controllers/controllers.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const {selectTopics} = require ('../Models/models')
+
+
+exports.getTopics = (req, res, next) => {
+selectTopics ()
+.then ((topics)=>{
+    res.status(200).send(topics)
```
Comment (roisinoreilly) [inline Controllers/controllers.js]: you need to send topics to the client on a meaningful key eg. {topics}

File: __tests__/app.test.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+});
+
+describe("3. GET /api/topics", () => {
+  test("the URL to respond with status 200 and the response object to contain an array of topics", () => {
+    return request(app)
+      .get('/api/topics')
+      .expect(200)
+      .then(({ body }) => {
+        expect(body).toBeInstanceOf(Array);
+        expect(body).toHaveLength(3);
```
Comment (roisinoreilly) [inline __tests__/app.test.js]: good job checking the length of the array before iterating through

File: app.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+const express = require("express");
+const app = express();
+const { getTopics } = require("./Controllers/controllers");
+const {
+  handle404Paths,
+  handle500Errors,
+} = require("./Controllers/controllers.errors");
+app.use(express.json());
```
Comment (roisinoreilly) [inline app.js]: can you remove this line until we're using it please

File: listen.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+const app = require('./app')
```
Comment (roisinoreilly) [inline listen.js]: we don't need any of this until ticket 14 and we start thinking about hosting, supertest has it's own built in listener

**Review by roisinoreilly — CHANGES_REQUESTED** [review]
Great job so far! Would you be able to send topics back in an object and remove the code I've commented that we're not using yet to keep everything strictly TDD. After doing that send us another quick PR :)

**Comment by robert112233** [comment]
Merge when ready 👍

**Pull request merged**